app/middleware/postcodeLookup.js

```diff
@@ -1,21 +1,43 @@
+const postcodeValidator = require('../lib/postcodeValidator');
+const log = require('../lib/logger');
 const PostcodesIO = require('postcodesio-client');
 
 const postcodes = new PostcodesIO();
 
 function lookupPostcode(req, res, next) {
-  const outcodeRegex = /^[A-Z]{1,2}[0-9][0-9A-Z]?$/gi;
-  const postcodeRegex = /^[A-Z]{1,2}[0-9][0-9A-Z]?\s?[0-9][A-Z]{2}$/gi;
   const search = res.locals.processedSearch;
 
-  if (search.match(outcodeRegex)) {
+  if (postcodeValidator.isOutcode(search)) {
     postcodes.outcode(search, (err, postcode) => {
```

Comment:
`postcode` is a confusing parameter name (? `outcodeDetails`). (sorry, I know it was me who named it!)

`postcodes` isn't great either. Maybe `const PostcodesIO = require('postcodesio-client')();` (with `postcodes` renamed) would be better.